### ISSUE TITLE
cli_inform doesn't invoke cli.default_handler

### ISSUE THREAD

**Issue by inferentialist** [r-lib/cli#712]
Based on [https://cli.r-lib.org/articles/semantic-cli.html#cli-messages](https://cli.r-lib.org/articles/semantic-cli.html#cli-messages), I would expect `cli_abort`, `cli_warn`, and `cli_inform` to call the handler specified in `cli.default_handler`.  However, these particular functions wrap their `rlang` analogues and seemingly wind up on another code path.

Here's a simple reproducible example.  The first `expect_output` behaves as expected; the second, fails.

```R
test_that("cli_inform doesn't call cli.default_handler", {

  test_case <- function(cli_op) {
    withr::with_options(
      list(
        cli.default_handler = function(msg) cat("handled")
      ),
      cli_op("foo")
    )
  }

  expect_output(
    test_case(cli::cli_alert),
    "handled"
  )

  expect_output(
    test_case(cli::cli_inform),
    "handled"
  )
})
```

Hoping this has an easy workaround; and apologies if I overlooked something in the documentation.  Thanks for all your work on the `cli` project!

**Comment by gaborcsardi** [comment]
Yes, that's a mistake in the documentation, those functions indeed do not use the regular cli code path.

**Comment by inferentialist** [comment]
Thanks for the clarification.  

Wondering if something like this might be a palatable alternative:

```R
  alt_inform <- function(message, ..., .envir = parent.frame()) {
    outer_cnd <- catch_cnd(
      rlang::inform(cli::format_message(message, .envir = .envir), ...)
    )

    cnd <- catch_cnd(
      cli::cli_verbatim(
        format(outer_cnd)
      )
    )

    handler <- getOption("cli.default_handler", cli:::cli_server_default)
    handler(cnd)

    cnd_signal(outer_cnd)
    invisible(NULL)
  }
```

**Comment by gaborcsardi** [comment]
`cli_inform()` is basically the same as `cli_bullets()`, so you can probably use that instead.

**Comment by inferentialist** [comment]
My hope was to use the default handler functionality to intercept cli_* calls and reroute them into a log.  In general, I wouldn't have access to the source code that invokes the cli_* calls.

Seems like it wasn't designed for this use case.  No worries.  Thanks for the comments.

**Comment by gaborcsardi** [comment]
You can always catch all messages:
```r
❯ msgs <- list(); withCallingHandlers(cli::cli_inform("good to know"), message = function(m) msgs <<- c(msgs, list(m)))
good to know

❯ msgs
[[1]]
<message/rlang_message>
Message:
good to know
```

**Comment by gaborcsardi** [comment]
See an example for catching all (formatted) messages (not just from cli) here: https://github.com/r-lib/cli/issues/708#issuecomment-2647706839
